MarvelData/Domain/Core/UsecaseResult.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  UsecaseResult.swift
+//  MarvelData
+//
+//  Created by Swapnali Patil on 22/04/22.
+//
+
+import Foundation
+
+public enum Result<Value> {
```

Comment:
why public? unless it is necessary? because all the the code resides in main app then we don't need to expose them.
It would have been different case if this code was in another framework.